src/Uno.UI/UI/Xaml/UIElement.RoutedEvents.cs

```diff
@@ -549,7 +549,10 @@ internal bool RaiseEvent(RoutedEvent routedEvent, RoutedEventArgs args)
 				{
 					if (!IsHandled(args) || handler.HandledEventsToo)
 					{
-						InvokeHandler(handler.Handler, args);
+						if (IsEnabledOverride())
```

Comment:
I think this is not valid and I'm really uncomfortable to this like that.

For instance when a `Control` gets disabled we raise some pointer events (depending of the Pressed/Over/Capture state - and soon also for dragging state) (https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Pointers.cs#L53 and https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Pointers.cs#L119

But this method is also used for all `RoutedEvent`, so I would suspect that some other events can still be triggered even if the control is disabled e.g. `LostFocus`.

For the pointers, the `IsEnabled` is checked here https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Pointers.cs#L182 which is used by the hit testing. Actually you not even receive the native events.

Reply:
I just validated, when a control gets disabled while the pointer is over we do receive a `PointerExited`, but no `PointerReleased`. Not tested but I would assume the same for `CaptureLost`.

BTW: I think that the behavior is already correct on Skia as we dispatch pointers by ourselves.

Reply:
100% agree with @dr1rrb here. The enability of a control should not be considered as a filter for events.

Reply:
Hm, that is weird, when I tried various pointer events over disabled button in UWP, it didn't trigger any of the `OnPointerXXX` or `OnManipulationXXX` methods, nor events. Maybe I made some mistake somewhere 🤔

Reply:
I actually noticed the issue on Skia, where a disabled button still raised all of the events, which caused issues in case of TabView

Reply:
Please see the sample page `RoutedEvents_IsEnabled.xaml` - there I have a custom button with an override for all routed events and whatever user does, none of them is triggered on UWP. So probably the case which needs to be handled differently is when `IsEnabled` changes? Or maybe I am missing something 😅

Reply:
Based on [docs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.control.isenabled?view=winrt-19041) it seems that indeed the control with `IsEnabled=false` cannot on itself raise input events:

> A disabled control is still visible to hit testing. It can't handle any input events on itself. However, a disabled control can still source the input events, and input routed events can bubble to a parent where they can be handled.

In this case, the bubbling should still apply, as the filter just prevents the events on this particular control from being raised, but the remainder of the logic remains. What do you think?

Reply:
Yes a disabled element should not raise pointer events, and yes the bubbling will still apply, but the issue is really the moment where we disable the control, we have to raise some events and a filter like this would mute them.

But actually I think the issue is really more that we should not even consider the element when we hit test to find the element on which we should raise the event. It's important that to properly ignore them in order to find element under disabled one for instance.

If you notice the issue on Skia then it's probably because the `HitVisibilityProperty` has not bee updated properly (or maybe just a bit to late) when the `IsEnabled` has changed. This should have been done here https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Pointers.Skia.cs#L573

I'll try to have a look to you sample soon

Reply:
Well after investigation, it appears that the fix you did [in `Control`](https://github.com/unoplatform/uno/pull/4185/files?diff=split&w=1#diff-7e87918895f8b453fa65dab2ed76539d91c6f5c9f896ef2962aaadd7af2e121aR58) was sufficient. We were validating the `IsEnabledOverride` in the coerced hit testability property, however as it was not override on skia, all element were always considered as enabled.

I however had add an update request on change on Skia: https://github.com/unoplatform/uno/pull/4185/files#diff-5420c68867b462946606ca55b247c946582f2d0b74028d4c47c9bd02c8eacdfdR79